loader.js

```diff
@@ -28,30 +29,37 @@ function finish() {
 
 async function CarmiLoader(loader) {
 	const callback = loader.async()
-	const statsPath = tempy.file({extension: 'json'})
 	const tempOutputPath = tempy.file({extension: 'js'})
 	const loaderOptions = loaderUtils.getOptions(loader) || {}
 
 	const options = {
 		source: loader.getDependencies()[0],
-		stats: statsPath,
 		format: 'cjs',
 		output: tempOutputPath,
 		debug: process.env.NODE_ENV !== 'production',
 		...loaderOptions
 	}
+
+    options.stats = getCacheFilePath({
+       fileType: 'stats',
+       path: options.source,
+       debug: options.debug,
+       format: options.format,
+       name: 'model'
+    });
```

Comment:
I mimicked the same behavior as the CLI so there's a deterministic path for every different Carmi run.